official/recommendation/ncf_keras_main.py

```diff
@@ -242,10 +244,13 @@ def run_ncf(_):
         cloning=params["clone_model_in_keras_dist_strat"])
 
     history = keras_model.fit(train_input_dataset,
+                              steps_per_epoch=num_train_steps,
                               epochs=FLAGS.train_epochs,
                               callbacks=[
                                   IncrementEpochCallback(producer),
                                   time_callback],
+                              validation_data=eval_input_dataset,
```

Comment:
Ideally, this is the only change that should be needed. I needed to add all the other things (steps etc) because of an odd bug in Keras. Can simplify once that bug is fixed.